plugin/ochttp/client_test.go

```diff
@@ -119,3 +120,115 @@ func TestClient(t *testing.T) {
 		}
 	}
 }
+
+func BenchmarkTransportNoInstrumentation(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoStats: true, NoTrace: true})
+}
+
+func BenchmarkTransportTraceOnly(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoStats: true})
+}
+
+func BenchmarkTransportStatsOnly(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{NoTrace: true})
+}
+
+func BenchmarkTransportAllInstrumentation(b *testing.B) {
+	benchmarkClientServer(b, &ochttp.Transport{})
+}
+
+// Copied from net/http/serve_test.go
+func benchmarkClientServer(b *testing.B, transport http.RoundTripper) {
+	b.ReportAllocs()
+	b.StopTimer()
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(rw, "Hello world.\n")
+	}))
+	defer ts.Close()
+	var client http.Client
+	client.Transport = transport
+	b.StartTimer()
```

Comment:
Why not just b.ResetTimer() and remove the b.StopTimer() above?

Reply:
I don't know, this code was copied from the standard library HTTP tests so didn't really want to change it

Reply:
I mean but we can fix that though, even stdlib code could use such fixes :)

Reply:
I guess I'm not convinced that ResetTimer is better? I don't really care TBH but I also don't want people to see the comment above that says "copied from ..." and wonder why I changed this - it might cause them to think there was some reason for it.